src/Accelerators/NNPA/NNPAAccelerator.hpp

```diff
@@ -21,14 +21,22 @@ namespace accel {
 namespace nnpa {
 
 class NNPAAccelerator final : public Accelerator {
-private:
-  static bool initialized;
+  friend Accelerator;
 
 public:
-  NNPAAccelerator();
+  ~NNPAAccelerator() {}
+
+  /// Define classof to be able to use isa<>, cast<>, dyn_cast<>, etc.
+  static bool classof(const Accelerator *accel) {
+    return accel->getKind() == Accelerator::Kind::NNPA;
+  }
+  static bool classof(const NNPAAccelerator *) { return true; }
+
+private:
+  NNPAAccelerator() : Accelerator(Accelerator::Kind::NNPA) {}
 
-  bool isActive() const final;
```

Comment:
isActive not needed because each accelerator is constructed via a factory method and is immediately active upon construction (that is initialized fully, all dialects it needs are registered etc..)